### PR TITLE
Implement Search within Select Dropdown to match StyleGuide #minor

### PR DESCRIPTION
**Pull request by reedsa**
Adds Search input inside dropdown menu for Select inputs.

Screen Shot 2019-04-22 at 12 30 14 PM

## Merge Checklist

- [X] Includes link to associated Github Issue or JIRA ticket
- [X] Has descriptive PR title and body
- [X] All changes are backwards compatible
- [x] QA'd by the SendGrid Style Guide designer
- [x] Code reviewed by a SendGrid frontend lead

### CODE REVIEW

File: src/select/select.module.scss [hunk 1]
```diff
@@ -225,3 +231,26 @@
...
+
+.input-search-wrap {
+  position: absolute;
+  width: 100%;
+  z-index: 999;
+  background: $white;
+  box-shadow: 0 2px 6px 0 rgba(41, 70, 97, 0.2);
+
+  :global(.search__control) {
+    margin: 18px 18px 22px;
```
Comment (richbachman) [inline src/select/select.module.scss]: Let's make these variables. If we can drop the bottom-margin by 2px also, we can use a variable for that.

```suggestion
    margin: $scale-base-up-03 $scale-base-up-03 $scale-base-up-04;
```

File: src/select/select.module.scss [hunk 2]
```diff
@@ -35,8 +35,14 @@
     top: 5px;
   }
 
+  .input-text-wrap {
```
Comment (chrischalstrom) [inline src/select/select.module.scss]: might want to add these rules to styleguide?

Comment (reedsa) [reply src/select/select.module.scss]: Actually I didn't end up needing these, I'll remove them.

**Comment by thinkingserious** [comment]
Hello @reedsa,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**